### PR TITLE
DashboardScene: Support for dashboard PanelContext actions via state hook

### PR DESCRIPTION
**Pull request by torkelo**
In this PR we instead leverage a new VizPanel state function extendPanelContext (does not exist yet but you this is called from VizPanel buildPanelContext 

* [x] canAddAnnotations + tests
* [x] canEditAnnotations + tests
* [x] canDeleteAnnotation + tests
* [x] onAnnotationCreate + tests
* [x] onAnnotationDelete + tests
* [x] onAnnotationUpdate + tests
* [x] onAddAdHocFilter + tests

### CODE REVIEW

**Comment by dprokop** [comment]
As we discussed - this is the way to go with the context instead of https://github.com/grafana/grafana/pull/76173 and https://github.com/grafana/scenes/pull/397 which has the problem of typing the extended scene objecs as the type of the object being extended.

File: public/app/features/dashboard-scene/scene/setDashboardPanelContext.ts [hunk 1]
```diff
@@ -0,0 +1,188 @@
...
+      isRegion,
+      time: event.from,
+      timeEnd: isRegion ? event.to : 0,
+      tags: event.tags,
+      text: event.description,
+    };
+
+    await saveAnnotation(anno);
+
+    reRunBuiltInAnnotationsLayer(dashboard);
```
Comment (dprokop) [inline public/app/features/dashboard-scene/scene/setDashboardPanelContext.ts:71]: This turns out really nice!

**Review by dprokop — APPROVED** [review]
Looks nice and clean and works 💪

File: public/app/features/dashboard-scene/scene/setDashboardPanelContext.ts (hunk 1 above)
Comment (torkelo) [reply public/app/features/dashboard-scene/scene/setDashboardPanelContext.ts:71]: @dprokop I agree! I was really happy you had added that function "runLayer" :heart:

**Pull request merged**